Remove experimental annotation from Beam

kennknowles opened the pull request:
Drop @Experimental annotations and concept from Beam. Discussion occurred at https://lists.apache.org/thread/tvvdckdom8jtv2xr9mzg0ltjjpbmydrv.

Largely automated change. There are likely mentions in comments and elsewhere that will have follow-up changes to clean up the conceptual framework.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[![Build python source distribution and wheels](https://github.com/apache/beam/workflows/Build%20python%20source%20distribution%20and%20wheels/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[![Python tests](https://github.com/apache/beam/workflows/Python%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[![Java tests](https://github.com/apache/beam/workflows/Java%20Tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[![Go tests](https://github.com/apache/beam/workflows/Go%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

brucearctor commented on model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto:
```diff
@@ -1059,12 +1058,11 @@ message StandardCoders {
     // Example of well known logical types:
     //   beam:logical_type:schema:v1
-    //   - Representation type: BYTES
+    //   - Representation type: BYTE
```
Doesn't seem related to the purpose of the PR -- but OK anyways

brucearctor commented on sdks/python/apache_beam/utils/annotations_test.py:
```diff
@@ -101,51 +100,6 @@ def fnc_test_deprecated_without_since_custom_should_fail():
-
-      @experimental(current='multiply', extra_message='Do this')
-      def fnc_test_experimental_with_current_message():
-        return 'lol'
```
lol

brucearctor commented on sdks/python/apache_beam/utils/annotations.py:
```diff
@@ -192,10 +190,4 @@ def inner(*args, **kwargs):
-    current=None,
-    since=None,
-    extra_message=None)
+    annotate, label='deprecated', current=None, extra_message=None)
```
do we not care about newlines at end of file?

kennknowles replied:
I just ran `yapf` over the whole codebase according to https://cwiki.apache.org/confluence/display/BEAM/Python+Tips#PythonTips-Runningyapfformattermanually

brucearctor replied:
makes sense, then isn't part of the config -- definitely don't want to be nitpicky [ manually ] about such things.

brucearctor reviewed (approved):
LGTM [ once/if all tests pass ]

Looked at all the changes in the PR, and seems good.  I didn't check codebase [ ex: outside of this PR ] for whether this addressed ALL uses of experimental.

brucearctor commented:
Run Java PreCommit

brucearctor commented:
Run Java_Hbase_IO_Direct PreCommit

brucearctor commented:
Run Java_IOs_Direct PreCommit

brucearctor commented:
Run PythonLint PreCommit

brucearctor commented:
Run SQL PreCommit

brucearctor commented:
Run SQL_Java17 PreCommit

dspasic commented on runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowProfilingOptions.java:
```diff
@@ -18,14 +18,12 @@
 import org.apache.beam.sdk.options.PipelineOptions;
 
 /** Options for controlling profiling of pipeline execution. */
 @Description("[Experimental] Used to configure profiling of the Dataflow pipeline")
```
I would suggest to remove the experimental from the description

dspasic reviewed (commented):
Thanks for effort. I left some comments. Looking forward to your feedback.

jrmccluskey commented:
assign to next reviewer

kennknowles commented:
I did another pass and eliminated a very large number of occurrences in comments. I also reverted a couple weird changes that did not belong, and added a note to CHANGES.md

Abacn commented:
Test failure related:

```
19:29:10 > Task :sdks:java:extensions:sql:zetasql:buildDependents
19:29:10 
19:29:10 FAILURE: Build failed with an exception.
19:29:10 
19:29:10 * What went wrong:
19:29:10 Execution failed for task ':sdks:java:extensions:sql:jdbc:analyzeClassesDependencies'.
19:29:10 > Dependency analysis found issues.
19:29:10   unusedDeclaredArtifacts
19:29:10    - beam.sdks.java:core:2.48.0-SNAPSHOT@jar
```

it appears that extensions-sql-setasql no longer directly depends on java core

and a couple of whitespace and spotless

kennknowles commented:
- Fixed Python lint manually. I guess `yapf` doesn't automatically fix it :-/
- Removed whitespace from `CHANGES.md`
- Ran `spotlessApply`
- Fixed deps until `analyzeClassesDependencies` for SQL but having some weird stuff in the Kafka IT area:

```
* What went wrong:
Execution failed for task ':sdks:java:io:kafka:kafka-01103:analyzeClassesDependencies'.
> Dependency analysis found issues.
  unusedDeclaredArtifacts
   - beam.sdks.java.io:kafka:2.48.0-SNAPSHOT@jar
   - org.apache.kafka:kafka-clients:2.4.1@jar
 ```

It also seems we are generating gradle projects that do not match directories. This is going to cause a real headache in the future since it is not intended usage.

kennknowles commented:
Failures, from java precommit:

 - [org.apache.beam.runners.flink.FlinkRequiresStableInputTest.testParDoRequiresStableInputPortable](https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/27103/testReport/junit/org.apache.beam.runners.flink/FlinkRequiresStableInputTest/testParDoRequiresStableInputPortable/)
 - [org.apache.beam.runners.flink.FlinkRequiresStableInputTest.testParDoRequiresStableInputStatefulPortable](https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/27103/testReport/junit/org.apache.beam.runners.flink/FlinkRequiresStableInputTest/testParDoRequiresStableInputStatefulPortable/)
 - [org.apache.beam.sdk.util.UnboundedScheduledExecutorServiceTest.testThreadsAreAddedOnlyAsNeededWithContention](https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/27103/testReport/junit/org.apache.beam.sdk.util/UnboundedScheduledExecutorServiceTest/testThreadsAreAddedOnlyAsNeededWithContention/)

They all look like unrelated flakes to me.

kennknowles commented:
run java precommit

kennknowles commented:
Java_Spark3_Versions failed with gradle executor dying

kennknowles commented:
Run SQL_Java17 PreCommit

Abacn reviewed (approved):
Thanks, LGTM

kennknowles commented:
Resolved conflicts and pushed. So all the tests need to re-run anyhow. The problem was a known timeout in the GitHub Actions workflows. We will see if we can get it green.

kennknowles commented:
[org.apache.beam.sdk.io.gcp.spanner.SpannerReadIT.testReadWithDataBoost](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Commit/2569/testReport/junit/org.apache.beam.sdk.io.gcp.spanner/SpannerReadIT/testReadWithDataBoost/)

kennknowles commented:
@darshan-sj saw the above failure in Spanner databoost and I noticed you authored #26007

kennknowles commented:
Run Java_GCP_IO_Direct PreCommit

kennknowles commented:
[org.apache.beam.sdk.io.gcp.bigquery.BigQueryIOWriteTest.testTriggeredFileLoadsWithTempTablesToExistingNullSchemaTable[1]](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Phrase/147/testReport/junit/org.apache.beam.sdk.io.gcp.bigquery/BigQueryIOWriteTest/testTriggeredFileLoadsWithTempTablesToExistingNullSchemaTable_1_/)

kennknowles commented:
Run Java_GCP_IO_Direct PreCommit

kennknowles commented:
[org.apache.beam.sdk.io.gcp.spanner.SpannerReadIT.testReadWithDataBoost](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Phrase/148/testReport/junit/org.apache.beam.sdk.io.gcp.spanner/SpannerReadIT/testReadWithDataBoost/)

kennknowles commented:
Run Java_GCP_IO_Direct PreCommit

kennknowles commented:
I think we've kicked the flakes enough times here.
